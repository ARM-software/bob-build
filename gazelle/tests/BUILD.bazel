load("@bazel_gazelle//:def.bzl", "gazelle_generation_test")

[gazelle_generation_test(
    name = file[0:-len("/WORKSPACE")],
    size = "small",
    gazelle_binary = "//:gazelle_bin",
    test_data = glob(
        include = [file[0:-len("/WORKSPACE")] + "/**"],
    ),
) for file in glob(["**/WORKSPACE"])]
