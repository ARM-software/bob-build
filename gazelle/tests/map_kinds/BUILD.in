# gazelle:bob_root
# gazelle:map_kind config_setting my_config_setting //tools/bob:def.bzl
