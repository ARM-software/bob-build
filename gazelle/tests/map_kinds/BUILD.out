load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
load("//tools/bob:def.bzl", "my_config_setting")

# gazelle:bob_root
# gazelle:map_kind config_setting my_config_setting //tools/bob:def.bzl

my_config_setting(
    name = "[__FEATURE_A]",
    flag_values = {"//:FEATURE_A": True},
)

my_config_setting(
    name = "![FEATURE_A]",
    flag_values = {"//:FEATURE_A": False},
)

alias(
    name = "[FEATURE_A]",
    actual = ":[__FEATURE_A]",
)

my_config_setting(
    name = "[__FEATURE_B]",
    flag_values = {"//:FEATURE_B": True},
)

my_config_setting(
    name = "![FEATURE_B]",
    flag_values = {"//:FEATURE_B": False},
)

alias(
    name = "[FEATURE_B]",
    actual = ":[__FEATURE_B]",
)

bool_flag(
    name = "FEATURE_A",
    build_setting_default = False,
)

bool_flag(
    name = "FEATURE_B",
    build_setting_default = True,
)
